### PR TITLE
Pathfinding improvements

### PR DESCRIPTION
**Pull request by Earthcomputer**
Wow, there's some very low intermediary names in here

### CODE REVIEW

File: mappings/net/minecraft/block/Block.mapping [hunk 1]
```diff
@@ -113,7 +113,7 @@ CLASS net/minecraft/class_2248 net/minecraft/block/Block
 		ARG 4 random
 	METHOD method_9515 appendProperties (Lnet/minecraft/class_2689$class_2690;)V
 		ARG 1 builder
-	METHOD method_9516 canPlaceAtSide (Lnet/minecraft/class_2680;Lnet/minecraft/class_1922;Lnet/minecraft/class_2338;Lnet/minecraft/class_10;)Z
+	METHOD method_9516 canPathfindThrough (Lnet/minecraft/class_2680;Lnet/minecraft/class_1922;Lnet/minecraft/class_2338;Lnet/minecraft/class_10;)Z
```
Comment (liach) [inline mappings/net/minecraft/block/Block.mapping:116]: is `place at side` not a usage of this method?

File: mappings/net/minecraft/entity/ai/pathing/EntityNavigation.mapping [hunk 2]
```diff
@@ -3,18 +3,29 @@ CLASS net/minecraft/class_1408 net/minecraft/entity/ai/pathing/EntityNavigation
...
+	FIELD field_6672 pathStartPos Lnet/minecraft/class_243;
 	FIELD field_6673 pathNodeNavigator Lnet/minecraft/class_13;
+	FIELD field_6674 pathStartTime I
 	FIELD field_6675 tickCount I
 	FIELD field_6677 world Lnet/minecraft/class_1937;
 	FIELD field_6678 nodeMaker Lnet/minecraft/class_8;
 	FIELD field_6679 shouldRecalculate Z
+	FIELD field_6680 lastNodePosition Lnet/minecraft/class_243;
 	FIELD field_6681 currentPath Lnet/minecraft/class_11;
+	FIELD field_6682 currentNodeTimeout D
```
Comment (liach) [inline mappings/net/minecraft/entity/ai/pathing/EntityNavigation.mapping:18]: hmm, is the timeout really a double than a long

File: mappings/net/minecraft/block/Block.mapping (hunk 1 above)
Comment (liach) [reply mappings/net/minecraft/block/Block.mapping:116]: After a dive into I think this is just for goals and pathing. `BlockPlacementEnvironment` deserves a rename as well.

File: mappings/net/minecraft/entity/ai/pathing/EntityNavigation.mapping (hunk 2 above)
Comment (liach) [reply mappings/net/minecraft/entity/ai/pathing/EntityNavigation.mapping:18]: Yes this is, mojang is just weird.

**Review by liach — APPROVED** [review]

**Review by Runemoro — APPROVED** [review]

**Pull request merged**